packages/grafana-llm-app/src/components/AppConfig/DevSandbox/DevSandboxToolInspector.tsx

```diff
@@ -262,12 +386,43 @@ interface DevSandboxToolInspectorProps {
 export function DevSandboxToolInspector({ tools }: DevSandboxToolInspectorProps) {
   const [searchFilter, setSearchFilter] = useState('');
 
-  const filteredTools = tools.filter((tool) => {
+  // Memoize filtered tools to prevent recalculation on every render
+  const filteredTools = React.useMemo(() => {
     const searchLower = searchFilter.toLowerCase();
-    const name = (tool.annotations?.title ?? tool.name).toLowerCase();
-    const description = (tool.description || '').toLowerCase();
-    return name.includes(searchLower) || description.includes(searchLower);
-  });
+    return tools.filter((tool) => {
+      const name = (tool.annotations?.title ?? tool.name).toLowerCase();
+      const description = (tool.description || '').toLowerCase();
+      return name.includes(searchLower) || description.includes(searchLower);
```

Comment:
this should search the title, name _and_ description (rather than name OR title) - i tried searching for `list_datasources` and got nothing which was confusing.